src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.cpp

```diff
@@ -98,135 +106,207 @@ void jit_emitter::emitter_preamble(const std::vector<size_t>& in_idxs,
         OPENVINO_THROW("Failed to allocate required number of gpr registers");
     }
 
+    using namespace Xbyak_aarch64::util;
+    const bool is_vec_input = (in_out_type_ == emitter_in_out_map::vec_to_vec) ||
+                              (in_out_type_ == emitter_in_out_map::vec_to_gpr);
+    const bool is_vec_output = (in_out_type_ == emitter_in_out_map::vec_to_vec) ||
+                               (in_out_type_ == emitter_in_out_map::gpr_to_vec);
+
+    // vector registers
     for (auto idx : pool_aux_vec_idxs) {
         aux_vec_idxs.push_back(static_cast<uint32_t>(idx));
     }
 
+    for (size_t idx = 0; idx < get_max_vecs_count(); idx++) {
+        if (aux_vec_idxs.size() >= get_aux_vecs_count()) break;
+
+        if (is_vec_input) {
+            if (std::find(in_idxs.begin(), in_idxs.end(), idx) != in_idxs.end()) continue;
+        }
+        if (is_vec_output) {
+            if (std::find(out_idxs.begin(), out_idxs.end(), idx) != out_idxs.end()) continue;
+        }
+
+        if (std::find(in_idxs.begin(), in_idxs.end(), idx) != in_idxs.end()) continue;
+        if (std::find(out_idxs.begin(), out_idxs.end(), idx) != out_idxs.end()) continue;
+
+        if (std::find(aux_vec_idxs.begin(), aux_vec_idxs.end(), idx) != aux_vec_idxs.end()) continue;
+        if (std::find(preserved_vec_idxs.begin(), preserved_vec_idxs.end(), idx) != preserved_vec_idxs.end()) continue;
+
+        aux_vec_idxs.push_back(idx);
+        preserved_vec_idxs.push_back(idx);
+    }
+    if (aux_vec_idxs.size() < get_aux_vecs_count())
+        OV_CPU_JIT_EMITTER_THROW("Failed to allocate required number of vector registers");
+
+    // gpr registers
     for (auto idx : pool_aux_gpr_idxs) {
-        aux_gpr_idxs.push_back(static_cast<uint32_t>(idx));
-        preserved_gpr_idxs.push_back(static_cast<uint32_t>(idx));
+        aux_gpr_idxs.push_back(idx);
+    }
+
+    const uint32_t end_gpr_idx = Xbyak_aarch64::Operand::X30;
+    for (size_t gpr_idx = 0; gpr_idx <= end_gpr_idx; ++gpr_idx) {
+        size_t _idx = end_gpr_idx - gpr_idx; // we allocate from the end
+
+        if (aux_gpr_idxs.size() >= get_aux_gprs_count()) break;
+        if ((_idx == Xbyak_aarch64::Operand::X18) ||
```

Comment:
Could you please clarify why these registers cannot bu used? Worth to add comment with the description. CAn be done in separate PR